src/invert/laplace/impls/serial_band/serial_band.cxx

```diff
@@ -106,21 +106,19 @@ const FieldPerp LaplaceSerialBand::solve(const FieldPerp &b, const FieldPerp &x0
     xend = mesh->LocalNx-2;
   }
 
-  for(int iz=0;iz<=ncz/2;iz++) {
+  for(int iz=0;iz<=maxmode;iz++) {
```

Comment:
Is the value of the mode the result (xk1d?) set to zero anywhere for iz>maxmode?